test/core/providers/local-accounts.ts

```diff
@@ -0,0 +1,135 @@
+import { assert } from "chai";
+import Transaction from "ethereumjs-tx";
+import { EventEmitter } from "events";
+
+import { getConfig } from "../../../src/core/config";
+import { IEthereumProvider } from "../../../src/core/providers/ethereum";
+import {
+  createLocalAccountsProvider,
+  hashTransaction,
+  signTransaction
+} from "../../../src/core/providers/local-accounts";
+import { useFixtureProject } from "../../helpers/project";
+
+class MockProvider extends EventEmitter implements IEthereumProvider {
+  public async send(method: string, params?: any[]): Promise<any> {
+    if (method === "eth_getTransactionCount") {
+      return 0x08;
+    }
+    if (method === "eth_sendRawTransaction") {
+      if (params !== undefined) {
+        return params[0];
+      }
+    }
+  }
+}
+
+describe("ethereum provider", () => {
+  let mock: MockProvider;
+  let wrapper: IEthereumProvider;
+
+  let accounts: string[] = [];
+  const chainId = 3;
+  useFixtureProject("config-project");
+
+  beforeEach(() => {
+    const [config, _] = getConfig();
+    accounts = config.networks.develop.accounts;
+    mock = new MockProvider();
+    wrapper = createLocalAccountsProvider(mock, accounts, chainId);
+  });
+
+  it("eth_accounts", async () => {
+    const response = await wrapper.send("eth_accounts");
+    // This should check that the address is derivated from the private key
+    assert.equal(response[0].length, 20);
+  });
+
+  it("eth_requestAccounts", async () => {
+    const response = await wrapper.send("eth_requestAccounts");
+    assert.equal(response[0].length, 20);
+  });
+
+  it("eth_sign", () => {
+    wrapper
+      .send("eth_sign")
+      .then(() => {})
+      .catch(err => assert.equal(err.message, "eth_sign is not supported yet"));
+  });
+
+  it("sendTransaction without specify gas", async () => {
+    const params = [
+      {
+        from: "0xf7abeea1b1b97ef714bc9a118b0f095ec54f8221",
+        to: "0x2a97a65d5673a2c61e95ce33cecadf24f654f96d",
+        gasPrice: 0x3b9aca00,
+        nonce: 0x8,
+        chainId: 3
+      }
+    ];
+
+    wrapper
+      .send("eth_sendTransaction", params)
+      .then(() => {})
```

Comment:
This test won't fail if nothing is thrown. See #102 for a helper that handles this kind of tests.